cpp/ql/src/semmle/code/cpp/models/implementations/Iterator.qll

```diff
@@ -157,9 +157,11 @@ private class IteratorSubOperator extends Operator, TaintFunction {
  * A non-member `operator+=` or `operator-=` function for an iterator type.
  */
 private class IteratorAssignArithmeticOperator extends Operator, DataFlowFunction, TaintFunction {
+  FunctionInput iteratorInput;
+
   IteratorAssignArithmeticOperator() {
     this.hasName(["operator+=", "operator-="]) and
-    this.getDeclaringType() instanceof Iterator
+    iteratorInput = getIteratorArgumentInput(this, 0)
```

Comment:
The new field isn't referenced anywhere else - should it be deleted or did you plan to use it somewhere else?

Reply:
Good point, I've removed it now.